apps/workflows/jobs/refill-daily.ts

```diff
@@ -13,29 +13,52 @@ client.defineJob({
   }),
 
   run: async (_payload, io, _ctx) => {
+    const date = _payload.ts;
+    const lastDayOfMonth = new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate();
+    const today = date.getDate();
     const db = connectDatabase();
     const tb = new Tinybird(env().TINYBIRD_TOKEN);
-    const t = new Date();
-    t.setUTCHours(t.getUTCHours() - 24);
+    let keys: Key[];
+    console.log("today", today);
+    console.log("last day of month", lastDayOfMonth);
 
-    const keys = await io.runTask("list keys", () =>
-      db.query.keys.findMany({
-        where: (table, { isNotNull, isNull, eq, and, gt, or }) =>
-          and(
-            isNull(table.deletedAt),
-            isNotNull(table.refillInterval),
-            isNotNull(table.refillAmount),
-            eq(table.refillInterval, "daily"),
-            gt(table.refillAmount, table.remaining),
-            or(
-              isNull(table.lastRefillAt),
-              lte(table.lastRefillAt, t), // Check if more than 24 hours have passed
-            ),
-          ),
-      }),
-    );
-    io.logger.info(`found ${keys.length} keys with daily refill set`);
-    // const keysWithRefill = keys.length;
+    if (today !== lastDayOfMonth) {
```

Comment:
I had to wrap my head around how this works
it seems to be correct though

can we make it a bit easier to understand?
Maybe it's just the time though, I'm tired